pkg/helper/loki_config.go

```diff
@@ -0,0 +1,102 @@
+package helper
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	flowslatest "github.com/netobserv/network-observability-operator/api/v1beta2"
+)
+
+type LokiConfig struct {
+	flowslatest.LokiManualParams
+	BatchWait    *metav1.Duration
+	BatchSize    int64
+	Timeout      *metav1.Duration
+	MinBackoff   *metav1.Duration
+	MaxBackoff   *metav1.Duration
+	MaxRetries   *int32
+	StaticLabels map[string]string
+}
+
+func NewLokiConfig(spec *flowslatest.FlowCollectorLoki) LokiConfig {
+	loki := LokiConfig{
+		BatchWait:    spec.BatchWait,
+		BatchSize:    spec.BatchSize,
+		Timeout:      spec.Timeout,
+		MinBackoff:   spec.MinBackoff,
+		MaxBackoff:   spec.MaxBackoff,
+		MaxRetries:   spec.MaxRetries,
+		StaticLabels: spec.StaticLabels,
+	}
+	switch spec.Mode {
+	case flowslatest.LokiModeLokiStack:
+		dotNamespace := ""
+		if len(spec.LokiStack.Namespace) > 0 {
+			dotNamespace = "." + spec.LokiStack.Namespace
+		}
+		gatewayURL := fmt.Sprintf("https://%s-gateway-http%s.svc:8080/api/logs/v1/network/", spec.LokiStack.Name, dotNamespace)
```

Comment:
We could fallback on `FlowCollectorSpec.namespace` here to have something easier to read and consistent

Reply:
this would add a dependency on `spec.namespace` in the function input, which would break the logic in `Convert_v1beta2_FlowCollectorLoki_To_v1alpha1_FlowCollectorLoki` (https://github.com/netobserv/network-observability-operator/pull/474/files#diff-23e9a27bd609a9414ced41fdaf8063ade03a51b3154d85ad4edfdae228dd5531R139-R150) where we need to run this conversion only with `v1beta2.FlowCollectorLoki` as an input, not the whole spec. So removing the dependency on `spec.namespace` actually helps there.